### PR TITLE
[Spark] Override Jackson string length limits

### PR DESCRIPTION
**Pull request by olaky**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

Override the Jackson string length limit to allow reading and writing large json objects to and from the Delta Log.

## How was this patch tested?

Added a new test suite

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

File: spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    )
+    for ((obj, json) <- testCases) {
+      assert(JsonUtils.toJson(obj) == json)
+      assert(JsonUtils.fromJson[DataSize](json) == obj)
+    }
+  }
+
+  test("Serialize and de-serialize commit info with large message") {
+    val operationStringSize = StreamReadConstraints.DEFAULT_MAX_STRING_LEN * 10
+    assert(operationStringSize > StreamReadConstraints.DEFAULT_MAX_STRING_LEN)
```
Comment (c27kwan) [inline spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala:51]: Is this a bit redundant, since the variable is defined based on this constant?

File: spark/src/main/scala/org/apache/spark/sql/delta/util/JsonUtils.scala [hunk 2]
```diff
@@ -28,6 +29,13 @@ object JsonUtils {
     _mapper.setSerializationInclusion(Include.NON_ABSENT)
     _mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
     _mapper.registerModule(DefaultScalaModule)
+
```
Comment (c27kwan) [inline spark/src/main/scala/org/apache/spark/sql/delta/util/JsonUtils.scala:32]: Maybe add a quick comment about us overriding the default. (e.g. why int.max and not long.maxvalue). Was this the old default value?

File: spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import org.apache.spark.sql.test.SharedSparkSession
+
+class JsonUtilsSuite
+  extends QueryTest
+  with SharedSparkSession {
+
+  test("DataSize json serialization") {
+    val testCases = Seq(
+      DataSize() -> """{}""",
+      DataSize(bytesCompressed = Some(816L)) -> """{"bytesCompressed":816}""",
```
Comment (c27kwan) [inline spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala:37]: Can you add a bit more variance in the data by adding negative long and 0L?

Comment (c27kwan) [reply spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala:37]: the negative long is maybe more appropriate for the other test, since this one is data size, which shouldn't be negative.

File: spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala (hunk 1 above)
Comment (c27kwan) [reply spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala:51]: Did you mean to assert operationStringSize < Int.MaxValue (which can also be its own constant like Delta_JSON_COMMIT_MAX_String_LEN

Comment (olaky) [reply spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala:51]: What this is supposed to check is that when the value of DEFAULT_MAX_STRING_LEN changes, the *10 does not lead to an overflow because for this test to be relevant, operationStringSize has to be bigger than the default max (because we want to validate that we have a custom default)

File: spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala (hunk 3 above)
Comment (olaky) [reply spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala:37]: I added 0 and negative. While neagtive sizes are not valid, serialization should still handle them

File: spark/src/main/scala/org/apache/spark/sql/delta/util/JsonUtils.scala (hunk 2 above)
Comment (olaky) [reply spark/src/main/scala/org/apache/spark/sql/delta/util/JsonUtils.scala:32]: I added a comment. Since the parameter is of type int, I think commenting why it is not long.maxvalue is very redundant

**Review by c27kwan — APPROVED** [review]
lgtm! Thanks for fixing
